src/group.h

```diff
@@ -24,13 +24,14 @@ extern GroupPool _group_pool; ///< Pool of groups.
 /** Statistics and caches on the vehicles in a group. */
 struct GroupStatistics {
 	uint16 num_vehicle;                     ///< Number of vehicles.
+	Money profit_last_year;                 ///< Sum of profits for all vehicles.
 	uint16 *num_engines;                    ///< Caches the number of engines of each type the company owns.
 
 	bool autoreplace_defined;               ///< Are any autoreplace rules set?
 	bool autoreplace_finished;              ///< Have all autoreplacement finished?
 
 	uint16 num_profit_vehicle;              ///< Number of vehicles considered for profit statistics;
-	Money profit_last_year;                 ///< Sum of profits for all vehicles.
+	Money profit_last_year_min_age;         ///< Sum of profits for considered vehicles.
```

Comment:
Something to consider about the workings of C++. The actual data in this structures and classes are placed in the order they are written in the code. However, the access of (some) memory by the CPU is often much faster at multiples of their register size. For this the compiler just adds a bit of unused memory, so things get aligned again.

In this case, on x86-64 in the original code, there will be 6 unused bytes between `num_vehicle` and `num_engines` and another 4 bytes between `num_profit_vehicle` and `profit_last_year`. So, preferably the fields are from big to small, so there is no need to create those gaps.
```
struct GroupStatistics {
	Money profit_last_year;                 ///< Sum of profits for all vehicles.
	Money profit_last_year_min_age;         ///< Sum of profits for considered vehicles.
	uint16_t *num_engines;                  ///< Caches the number of engines of each type the company owns.
	uint16_t num_vehicle;                   ///< Number of vehicles.
	uint16_t num_profit_vehicle;            ///< Number of vehicles considered for profit statistics;
	bool autoreplace_defined;               ///< Are any autoreplace rules set?
	bool autoreplace_finished;              ///< Have all autoreplacement finished?
};
```
This uses the same amount of memory (on x86-64) but with an extra field for the money you want to store.